modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1645,6 +1645,52 @@ CV__DNN_INLINE_NS_BEGIN
                              float confThreshold = 0.5f, float nmsThreshold = 0.0f);
      };
 
+    /** @brief This class represents high-level API for YOLO object detection networks.
+      *
+      * YoloDetection allows to set params for preprocessing input image.
+      * YoloDetection creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return result detections.
+      */
+     class CV_EXPORTS_W_SIMPLE YoloObjectDetector : public DetectionModel
+     {
+     public:
+        /**
+          * @brief Create detection model from network represented in one of the supported formats.
+          * An order of @p model and @p config arguments does not matter.
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          */
+        //  CV_WRAP YoloObjectDetector(const String& model, const String& config = "");
+
+         /**
+          * @brief Create model from onnx graph.
+          * @param[in] onnx_path path to onnx graph.
+          */
+         CV_WRAP YoloObjectDetector(const String& onnx, const String& yoloName);
```

Comment:
Can we adopt this clas work with both Darknet and ONNX formats? Parent class can work with Darknet models so you can fallback to it.

Reply:
Oh, added by the latest commit, let me check